server/events/state_command_runner.go

```diff
@@ -0,0 +1,54 @@
+package events
+
+import (
+	"fmt"
+
+	"github.com/runatlantis/atlantis/server/events/command"
+)
+
+func NewStateCommandRunner(
+	pullUpdater *PullUpdater,
+	prjCmdBuilder ProjectStateCommandBuilder,
+	prjCmdRunner ProjectStateCommandRunner,
+) *StateCommandRunner {
+	return &StateCommandRunner{
+		pullUpdater:   pullUpdater,
+		prjCmdBuilder: prjCmdBuilder,
+		prjCmdRunner:  prjCmdRunner,
+	}
+}
+
+type StateCommandRunner struct {
+	pullUpdater   *PullUpdater
+	prjCmdBuilder ProjectStateCommandBuilder
+	prjCmdRunner  ProjectStateCommandRunner
+}
+
+func (v *StateCommandRunner) Run(ctx *command.Context, cmd *CommentCommand) {
+	var result command.Result
+	switch cmd.SubName {
+	case "rm":
+		result = v.runRm(ctx, cmd)
+	default:
+		result = command.Result{
+			Failure: fmt.Sprintf("unknown import subcommand %s", cmd.SubName),
+		}
+	}
+	v.pullUpdater.updatePull(ctx, cmd, result)
+}
+
+func (v *StateCommandRunner) runRm(ctx *command.Context, cmd *CommentCommand) command.Result {
+	projectCmds, err := v.prjCmdBuilder.BuildStateRmCommands(ctx, cmd)
+	if err != nil {
+		ctx.Log.Warn("Error %s", err)
+	}
+
+	if len(projectCmds) > 1 {
+		// There is no usecase to kick terraform state rm into multiple projects.
```

Comment:
What if you have multiple project names that use the same root module with different workspace names? If you have consistent resource names (which are not workspace dependent), then you could run `terraform state rm random_string.foo` across multiple projects, no?

Reply:
Thank you!